diff -up cronie-1.4.4/cronie.init.615107 cronie-1.4.4/cronie.init
--- cronie-1.4.4/cronie.init.615107	2011-02-18 10:08:25.896421180 +0100
+++ cronie-1.4.4/cronie.init	2011-02-18 10:09:17.089723595 +0100
@@ -44,11 +44,10 @@ start() {
     [ -x $exec ] || exit 5
     [ -f $config ] || exit 6
     echo -n $"Starting $prog: "
-    daemon $prog $CRONDARGS && success || failure
+    daemon $prog $CRONDARGS
     retval=$?
     echo
     [ $retval -eq 0 ] && touch $lockfile
-    return $retval
 }
 
 stop() {
@@ -66,7 +65,6 @@ stop() {
     retval=$?
     echo
     [ $retval -eq 0 ] && rm -f $lockfile
-    return $retval
 }
 
 restart() {
