diff -up cronie-1.4.4/src/crontab.c.temp-name cronie-1.4.4/src/crontab.c
--- cronie-1.4.4/src/crontab.c.temp-name	2015-09-21 14:09:02.629750993 +0200
+++ cronie-1.4.4/src/crontab.c	2015-09-21 14:10:01.291103910 +0200
@@ -621,7 +621,7 @@ static int replace_cmd(void) {
 	}
 
 	if (!glue_strings(TempFilename, sizeof TempFilename, SPOOL_DIR,
-			"tmp.XXXXXXXXXX", '/')) {
+	    "#tmp.XXXXXXXXXX", '/')) {
 		TempFilename[0] = '\0';
 		fprintf(stderr, "path too long\n");
 		return (-2);
