diff -up cronie-1.4.4/contrib/0anacron.675077 cronie-1.4.4/contrib/0anacron
--- cronie-1.4.4/contrib/0anacron.675077	2011-02-18 10:16:11.989741824 +0100
+++ cronie-1.4.4/contrib/0anacron	2011-02-18 10:16:21.272880899 +0100
@@ -1,6 +1,6 @@
 #!/bin/bash
 #in case file doesn't exist 
-if test -x /var/spool/anacron/cron.daily; then
+if test -r /var/spool/anacron/cron.daily; then
     day=`cat /var/spool/anacron/cron.daily`
 fi
 if [ `date +%Y%m%d` = "$day" ]; then
