diff -up cronie-1.4.11/src/cron.c.shutdown-msg cronie-1.4.11/src/cron.c
--- cronie-1.4.11/src/cron.c.shutdown-msg	2015-04-21 15:29:14.218651528 +0200
+++ cronie-1.4.11/src/cron.c	2015-04-21 15:29:30.212023639 +0200
@@ -486,6 +486,8 @@ int main(int argc, char *argv[]) {
 		log_it("CRON", pid, "INFO", "Inotify close failed", errno);
 #endif
 
+	log_it("CRON", pid, "INFO", "Shutting down", 0);
+
 	(void) unlink(_PATH_CRON_PID);
 
 	return 0;
