diff -up cronie-1.4.4/man/crontab.5.mail cronie-1.4.4/man/crontab.5
--- cronie-1.4.4/man/crontab.5.mail	2013-08-12 14:30:07.000000000 +0200
+++ cronie-1.4.4/man/crontab.5	2013-08-12 14:31:30.718994334 +0200
@@ -115,6 +115,12 @@ upper limit specified by the variable. T
 determined during the cron daemon startup so it remains constant for
 the whole run time of the daemon.
 .PP
+If the
+.I CRON_CORRECT_MAIL_HEADER
+environment variable is present regardless of its value, it will
+make crond to send e-mails with RFC2822 compliant From field. (Red Hat
+Enterprise Linux 6 only)
+.PP
 The format of a cron command is very much the V7 standard, with a number of
 upward-compatible extensions.  Each line has five time and date fields,
 followed by a user name if this is the system crontab file,
diff -up cronie-1.4.4/src/do_command.c.mail cronie-1.4.4/src/do_command.c
--- cronie-1.4.4/src/do_command.c.mail	2013-08-12 14:30:07.599809220 +0200
+++ cronie-1.4.4/src/do_command.c	2013-08-12 14:30:07.618809263 +0200
@@ -395,6 +395,7 @@ static void child_process(entry * e, use
 				char *content_type = env_get("CONTENT_TYPE", jobenv),
 					*content_transfer_encoding =
 					env_get("CONTENT_TRANSFER_ENCODING", jobenv);
+				char *p;
 
 				gethostname(hostname, MAXHOSTNAMELEN);
 
@@ -414,7 +415,13 @@ static void child_process(entry * e, use
 					(void) _exit(ERROR_EXIT);
 				}
 
-				fprintf(mail, "From: %s (Cron Daemon)\n", mailfrom);
+				p = env_get("CRON_CORRECT_MAIL_HEADER", jobenv);
+				if (p) {
+					fprintf(mail, "From: \"(Cron Daemon)\" <%s>\n", mailfrom);
+				}
+				else {
+					fprintf(mail, "From: %s (Cron Daemon)\n", mailfrom);
+				}
 				fprintf(mail, "To: %s\n", mailto);
 				fprintf(mail, "Subject: Cron <%s@%s> %s\n",
 					usernm, first_word(hostname, "."), e->cmd);
diff -up cronie-1.4.4/src/env.c.mail cronie-1.4.4/src/env.c
--- cronie-1.4.4/src/env.c.mail	2013-08-12 14:30:07.616809258 +0200
+++ cronie-1.4.4/src/env.c	2013-08-12 14:30:07.619809265 +0200
@@ -123,6 +123,7 @@ int env_set_from_environ(char ***envpp)
                "LC_ALL",
                "LANGUAGE",
                "RANDOM_DELAY",
+               "CRON_CORRECT_MAIL_HEADER",
                NULL
        };
        const char **name;
