diff -up cronie-1.4.4/src/popen.c.697485 cronie-1.4.4/src/popen.c
--- cronie-1.4.4/src/popen.c.697485	2013-05-23 17:15:41.899582557 +0200
+++ cronie-1.4.4/src/popen.c	2013-05-23 17:16:55.095574699 +0200
@@ -102,6 +102,9 @@ FILE *cron_popen(char *program, const ch
 			(void) close(pdes[1]);
 		}
 
+		if (cron_change_user_permanently(pw, pw->pw_dir) != 0)
+			_exit(2);
+
 		if (execvp(argv[0], argv) < 0) {
 			int save_errno = errno;
 
