diff -up cronie-1.4.4/src/misc.c.sigterm cronie-1.4.4/src/misc.c
--- cronie-1.4.4/src/misc.c.sigterm	2015-02-05 15:13:46.000000000 +0100
+++ cronie-1.4.4/src/misc.c	2015-02-05 15:26:39.910181647 +0100
@@ -285,6 +285,9 @@ void acquire_daemonlock(int closeflag) {
 			close(fd);
 			fd = -1;
 		}
+		/* and restore default sig handlers so we don't remove pid file if killed */
+		signal(SIGINT,SIG_DFL);
+		signal(SIGTERM,SIG_DFL);
 		return;
 	}
 
