diff -up cronie-1.4.4/cronie.init.init2 cronie-1.4.4/cronie.init
--- cronie-1.4.4/cronie.init.init2	2013-09-12 10:26:05.812117884 +0200
+++ cronie-1.4.4/cronie.init	2013-09-12 10:27:05.998273299 +0200
@@ -34,7 +34,7 @@ config=/etc/sysconfig/crond
 # Source function library.
 . /etc/rc.d/init.d/functions
 
-[ -e /etc/sysconfig/$prog ] && . /etc/sysconfig/$prog
+[ $UID -eq 0 ] && [ -e /etc/sysconfig/$prog ] && . /etc/sysconfig/$prog
 
 start() {
     if [ $UID -ne 0 ] ; then
@@ -68,7 +68,7 @@ stop() {
 }
 
 restart() {
-    stop
+    rh_status_q && stop
     start
 }
 
