diff -up cronie-1.4.4/contrib/0anacron.comments cronie-1.4.4/contrib/0anacron
--- cronie-1.4.4/contrib/0anacron.comments	2013-08-06 15:55:40.000000000 +0200
+++ cronie-1.4.4/contrib/0anacron	2013-08-06 16:02:53.643983115 +0200
@@ -1,5 +1,5 @@
 #!/bin/bash
-#in case file doesn't exist 
+# Skip excecution unless the date has changed from the previous run 
 if test -r /var/spool/anacron/cron.daily; then
     day=`cat /var/spool/anacron/cron.daily`
 fi
@@ -7,8 +7,7 @@ if [ `date +%Y%m%d` = "$day" ]; then
     exit 0;
 fi
 
-# in case anacron is already running,
-# there will be log (daemon won't be running twice).
+# Skip excecution unless AC powered
 if test -x /usr/bin/on_ac_power; then
     /usr/bin/on_ac_power &> /dev/null
     if test $? -eq 1; then
