diff -up cronie-1.4.4/src/crontab.c.676081 cronie-1.4.4/src/crontab.c
--- cronie-1.4.4/src/crontab.c.676081	2011-02-17 14:56:59.210863896 +0100
+++ cronie-1.4.4/src/crontab.c	2011-02-17 14:59:41.412874958 +0100
@@ -713,7 +713,7 @@ static int replace_cmd(void) {
 		goto done;
 	}
 
-	file_owner = (getgid() == getegid())? ROOT_UID : pw->pw_uid;
+	file_owner = (getgid() == geteuid() && getgid() == getegid()) ? ROOT_UID : pw->pw_uid;
 
 #ifdef HAS_FCHOWN
 	if (fchown(fileno(tmp), file_owner, -1) < OK) {
